### PR TITLE
remove the unnecessary "pkg" package level.

### PR DESCRIPTION
**Pull request by jmazzitelli**
Everything is a package, no need to have "pkg" at the top. It is superfluous.

Also, run gofmt - these files weren't formatted originally

### CODE REVIEW

**Comment by lucasponce** [comment]
Before to run this merge I suggest to open a little bit the discussion.
Two things:
- I completely forgot about the gofmt, I thought travis hook were in place and it should fail on style.
- About the "pkg", kubernetes, istio and prometheus use this way to fold and structure code, as using "cmd", or other conventions, it's not a problem to adapt to other conventions. It is not discussing about it.

**Comment by jmazzitelli** [comment]
set your IDE to gofmt on file save - its fast, and easier. Things are changed immediately and you won't have to worry about odd diffs that occur when you commit (assuming a git hook is even in place).

As for pkg package - I don't even know what the prometheus source code style is supposed to be - it is confusing what their pkg is compared to all the other packages in their go app - https://github.com/prometheus/prometheus

Why are some not in pkg (like the promql, notifier, rules, scrape packages) but then they have this other pkg package? It is confusing and why even call a package "pkg" (short for "package") in the first place when they have other packages not in pkg?  Seems odd to me, and certainly isn't obvious why you would put one package in pkg but another package not in pkg.

**Comment by jshaughn** [comment]
Yes, that is what I find confusing.  It's not a big deal though, we can bring it back.  I have a small PR I'm going to generate, I'll just add it back.

**Comment by lucasponce** [comment]
Thanks for the tip about gofmt, I will update my IDE, or add a local script to not forget before PR.

About code convention, I can't stand for any pro/cons, I just simply added what I think other projects were doing.

Kubernetes:
https://github.com/kubernetes/apimachinery/tree/master/pkg
https://github.com/kubernetes/kubernetes/tree/master/pkg

Istio:
https://github.com/istio/istio/tree/master/pkg
https://github.com/istio/istio/tree/master/mixer/pkg
https://github.com/istio/istio/tree/master/pilot/pkg

Jaeger:
https://github.com/jaegertracing/jaeger/tree/master/pkg

Prometheus:
https://github.com/prometheus/prometheus/tree/master/pkg

So, I saw a pattern and a convention that was my rationale to include it. My first thought is that layers that can be used as "services" in the abstraction concept can go there. That's why I introduce it, it made sense for the k8s and p8s ones.

Again, change is trivial, and minimal, we can add/remove it at any time, if this helps to decide what we want it's fine.

**Comment by jmazzitelli** [comment]
OK, I looked at all of those to see if I can understand what the common practice is for a "pkg" package. Perhaps there is some standard here? But after looking at those, there does not seem to be any common practice or standard other than to include a folder with the name "pkg" somewhere in their Go structure (and not all Go projects have these anyway).

Some "pkg" packages do not make sense or are not obvious (this is the Prometheus case - e.g. why does one thing go in pkg package but another go in the scrape package? https://github.com/prometheus/prometheus/tree/master/scrape or in any other packages like notifier, rules, etc.) So the pkg package in Prometheus seems confusing to me. I have no idea why something is in pkg but others are not. I am sure they have their reasons, but it isn't intuitively obvious (this is my one main problem with this "pkg" - it is a package called "package" (aka pkg) so it's purpose is mysterious without a more descriptive name).

Some projects are putting things in "pkg" but do not have a current need to do so. jaegar has pkg, but also has other top level packages (such as model https://github.com/jaegertracing/jaeger/tree/master/model ). They do this because it seems they want to use pkg as a kind of "util" dumping ground and they only do this because they may want to (in the future) put it all in a separate repo as their README says: https://github.com/jaegertracing/jaeger/blob/master/pkg/README.md . So to me this should be called "util" or "common" or something similar as it is more descriptive of what it is. It is weird to call a package "pkg" and have it used for only common utilities. That's like creating a Java util package called "pkg" :)

istio has a very complicated repo as described here - https://github.com/istio/istio#repositories
So their "pkg" folder is the full "istio" component code, and the other "package" folders aren't really Go source packages at all but rather sub-repos of additional go apps - see pilot and mixer as examples. This is a very complicated source structure that we do not have. We have a single go app container that doesn't follow the structure of the istio github repo. Their "pkg" is used in a different way than we would be using it. If we were to create a pkg directory at our root directory, then that "pkg" folder would just be the name of yet another Go package in a series of top level packages (e.g. pkg, handlers, routing, server, etc) and not how istio is using "pkg". In fact, the way we would do it would even violate istio's code conventions documented here https://github.com/istio/istio/blob/master/DEV-CONVENTIONS.md#directory-and-file-conventions where they say "Avoid general utility packages. Packages called "util" are suspect. Instead, derive a name that describes your desired function." In jaeger's case, they have it as an area to put common utils that may get refactored into their own repository in the future, and thus appears to violate these istio conventions. So I still question the use of the name "pkg" in Jaeger - why not use a more descriptive name like "util" or "common" or something other than package "pkg"?)

Finally, there are other Go projects out there that simply do not even have a pkg package - such as https://github.com/rhysd/gocaml , https://github.com/go-yaml/yaml , and others.

So, in short, there does not appear to be any common usage pattern for a folder called "pkg" in GoLang. They may be used in different ways, or some may not have pkg at all. And even some projects that use it do so in a less-than obvious way. In other words, there doesn't seem to be any single convention here other than there is this folder with a name called "pkg" that some, not all, Go projects have.

**Comment by lucasponce** [comment]
The discussion is interesting, but next time, let's do in first place instead to merge.

I think that pkg is used as a mere way to group things, and even there is not any IEEE strong convention, about it, it is like other languages when you can decide how to name/group things.

I guess that in case of doubt, the effective step is to propose something (some style, like a DEV-CONVENTION file, or even simpler, in the README a paragraph saying how we propose to organize).
Then, it will help others to understand what we follow and we will mimic same rules as close as we can.
(Without seeing that a following PR deletes some code without a clear explanation).
